internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -1216,10 +1219,103 @@ func (c *Coordinator) generateComponentModel() (err error) {
 	// If we made it this far, update our internal derived values and
 	// return with no error
 	c.derivedConfig = cfg
+	c.lastComponentModel = c.componentModel
```

Comment:
Do you actually need to save this? How big is it? Remember we can have agents with 100s of units.

Can you not just move the `c.componentModel = comps` to after this function call and pass in both the new and updated model as arguments? Like `checkAndLogUpdate(c.componentModel, comps)`?

Reply:
Ah, good idea.